diff -ur --new-file logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/convert.conf logitechmediaserver-server-7.8.0.1381491322/slimserver/convert.conf
--- logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/convert.conf	2013-03-02 22:55:32.000000000 +0000
+++ logitechmediaserver-server-7.8.0.1381491322/slimserver/convert.conf	2013-10-21 15:32:26.689393334 +0100
@@ -306,3 +306,25 @@
 
 spdr spdr * *
 	 -
+
+dsf dsf * *
+	# IFD
+	-
+
+dff dff * *
+	# IFD
+	-
+
+##
+## dsdplay DSD (dff) -> PCM (flac)
+##
+dff flc * *
+       # FT:{START=-s %t}U:{END=-e %v}D:{RESAMPLE=-r %d}
+       [dsdplay] $START$ $END$ $FILE$ $RESAMPLE$
+
+##
+## dsdplay DSD (dsf) -> PCM (flac)
+##
+dsf flc * *
+       # FT:{START=-s %t}U:{END=-e %v}D:{RESAMPLE=-r %d}
+       [dsdplay] $START$ $END$ $FILE$ $RESAMPLE$
diff -ur --new-file logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/Slim/Formats/DSD.pm logitechmediaserver-server-7.8.0.1381491322/slimserver/Slim/Formats/DSD.pm
--- logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/Slim/Formats/DSD.pm	1970-01-01 01:00:00.000000000 +0100
+++ logitechmediaserver-server-7.8.0.1381491322/slimserver/Slim/Formats/DSD.pm	2013-10-21 15:31:21.598381585 +0100
@@ -0,0 +1,52 @@
+package Slim::Formats::DSD;
+
+# Copyright (C) 2013 Kimmo <feedback@daphile.com> 
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License,
+# version 2.
+
+use strict;
+use base qw(Slim::Formats);
+
+use Audio::Scan;
+use Slim::Formats::MP3;
+use Slim::Utils::Log;
+
+sub getTag {
+	my $class = shift;
+	my $file  = shift || return {};
+	
+	my $s = Audio::Scan->scan( $file );
+	
+	my $info = $s->{info};
+	my $tags = $s->{tags};
+	
+	return unless $info->{song_length_ms};
+		
+	$tags->{SIZE}	    = $info->{audio_size};
+	$tags->{SECS}	    = $info->{song_length_ms} / 1000;
+	$tags->{RATE}	    = $info->{samplerate};
+	$tags->{CHANNELS}   = $info->{channels};
+	$tags->{SAMPLESIZE} = 1;
+	$tags->{LOSSLESS}   = 1;
+
+	if ( $info->{tag_diar_artist} ) { 
+	        $tags->{ARTIST} = $info->{tag_diar_artist}; 
+	}
+	if ( $info->{tag_diti_title} ) {
+	        $tags->{TITLE} = $info->{tag_diti_title};
+	} 
+
+	if ( $info->{id3_version} ) {
+	        $tags->{TAGVERSION} = $info->{id3_version};
+		Slim::Formats::MP3->doTagMapping($tags);
+	}
+	
+       return $tags;
+}
+
+*getCoverArt = \&Slim::Formats::MP3::getCoverArt;
+
+sub canSeek { 1 }
+
+1;
diff -ur --new-file logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/Slim/Formats.pm logitechmediaserver-server-7.8.0.1381491322/slimserver/Slim/Formats.pm
--- logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/Slim/Formats.pm	2013-03-02 22:55:32.000000000 +0000
+++ logitechmediaserver-server-7.8.0.1381491322/slimserver/Slim/Formats.pm	2013-10-21 15:26:50.744657869 +0100
@@ -64,6 +64,8 @@
 		'ape' => 'Slim::Formats::APE',
 		'wvp' => 'Slim::Formats::WavPack',
 		'ogf' => 'Slim::Formats::OggFLAC',
+		'dff' => 'Slim::Formats::DSD',
+		'dsf' => 'Slim::Formats::DSD',
 
 		# Playlist types
 		'asx' => 'Slim::Formats::Playlists::ASX',
diff -ur --new-file logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/strings.txt logitechmediaserver-server-7.8.0.1381491322/slimserver/strings.txt
--- logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/strings.txt	2013-10-11 12:52:09.000000000 +0100
+++ logitechmediaserver-server-7.8.0.1381491322/slimserver/strings.txt	2013-10-21 15:26:50.754657564 +0100
@@ -11822,6 +11822,38 @@
 	RU	Ogg FLAC
 	SV	Ogg FLAC
 
+DFF
+	CS	DFF
+	DA	DFF
+	DE	DFF
+	EN	DFF
+	ES	DFF
+	FI	DFF
+	FR	DFF
+	HE	DFF
+	IT	DFF
+	NL	DFF
+	NO	DFF
+	PL	DFF
+	RU	DFF
+	SV	DFF
+
+DSF
+	CS	DSF
+	DA	DSF
+	DE	DSF
+	EN	DSF
+	ES	DSF
+	FI	DSF
+	FR	DSF
+	HE	DSF
+	IT	DSF
+	NL	DSF
+	NO	DSF
+	PL	DSF
+	RU	DSF
+	SV	DSF
+
 MODIFIED
 	CS	Upraveno
 	DA	Ændret
diff -ur --new-file logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/types.conf logitechmediaserver-server-7.8.0.1381491322/slimserver/types.conf
--- logitechmediaserver-server-7.8.0.1381491322.orig/slimserver/types.conf	2013-03-02 22:55:32.000000000 +0000
+++ logitechmediaserver-server-7.8.0.1381491322/slimserver/types.conf	2013-10-21 15:26:50.754657564 +0100
@@ -16,7 +16,9 @@
 css     css             text/css                        - 
 cue     cue             audio/x-cue                     playlist 
 cur     -               audio/x-cue-referenced          -
-dir     -               application/directory           list 
+dff	dff		audio/dff			audio
+dir     -               application/directory           list
+dsf	dsf		audio/dsf			audio 
 dtd     dtd             application/xml-dtd             -
 flc     flac,flc,fla    audio/x-flac,audio/flac         audio
 fec     -               audio/x-cue-flac                playlist
--- logitechmediaserver-server-7.8.0.1386157128/slimserver/Slim/Player/Squeezebox.pm.orig	2013-12-09 14:38:08.080768878 +0000
+++ logitechmediaserver-server-7.8.0.1386157128/slimserver/Slim/Player/Squeezebox.pm	2013-12-09 14:42:45.592599713 +0000
@@ -750,6 +750,15 @@
 		$pcmchannels     = '?';
 		$outputThreshold = 0;
 
+	} elsif ($format eq 'dff' || $format eq 'dsf') {
+
+		$formatbyte      = 'd';
+		$pcmsamplesize   = '?';
+		$pcmsamplerate   = '?';
+		$pcmendian       = '?';
+		$pcmchannels     = '?';
+		$outputThreshold = 0;
+
 	} elsif ( $handler->isa('Slim::Player::Protocols::SqueezePlayDirect') ) {
 
 		# Format handled by squeezeplay to allow custom squeezeplay protocol handlers
